### PR TITLE
add supply chain usecases to terraform scan

### PR DESCRIPTION
**Pull request by schosterbarak**
1. scan module version
2. prevent usage of external data blocks (that can execute code in plan phase)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/terraform/checks/data/external/ExternalData.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+class ExternalData(BaseDataCheck):
+    def __init__(self) -> None:
+        name = 'Ensure terraform external data blocks runs vetted code'
+        id = "CKV_TF_DATA_EXTERNAL_1"
+        supported_data = ["external"]
+        categories = [CheckCategories.CONVENTION]
+        super().__init__(name=name, id=id, categories=categories, supported_data=supported_data)
+
+    def scan_data_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
```
Comment (gruebel) [inline checkov/terraform/checks/data/external/ExternalData.py:15]: so basically the check fails if it finds a `data.external` block, right?

File: tests/terraform/checks/data/external/example_external_data/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# pylint: skip-file
```
Comment (gruebel) [inline tests/terraform/checks/data/external/example_external_data/wrapper.py:1]:
```python
# flake8: noqa
```
I think that's what you need to add at the top of the file

File: checkov/terraform/checks/data/external/ExternalData.py (hunk 1 above)
Comment (schosterbarak) [reply checkov/terraform/checks/data/external/ExternalData.py:15]: correct

File: tests/terraform/checks/module/registry/test_ModuleCheck.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        report = Runner().run(root_folder=test_files_dir, runner_filter=RunnerFilter(checks=["CKV_TF_MODULE_1"]),external_checks_dir=[external_checks])
+        summary = report.get_summary()
+
+
+        self.assertEqual(summary["passed"], 2)
+        self.assertEqual(summary["failed"], 3)
+        self.assertEqual(summary["skipped"], 0)
+        self.assertEqual(summary["parsing_errors"], 0)
+
```
Comment (gruebel) [inline tests/terraform/checks/module/registry/test_ModuleCheck.py:23]:
```suggestion

        check = next(c for c in module_registry.checks["module"] if c.id == "CKV_TF_MODULE_1")
        module_registry.checks["module"].remove(check)
```

File: tests/terraform/checks/module/registry/test_ModuleCheck.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        report = Runner().run(root_folder=test_files_dir, runner_filter=RunnerFilter(checks=["CKV_TF_MODULE_2"]),external_checks_dir=[external_checks])
+        summary = report.get_summary()
+
+
+        self.assertEqual(summary["passed"], 1)
+        self.assertEqual(summary["failed"], 4)
+        self.assertEqual(summary["skipped"], 0)
+        self.assertEqual(summary["parsing_errors"], 0)
+
```
Comment (gruebel) [inline tests/terraform/checks/module/registry/test_ModuleCheck.py:37]:
```suggestion

        check = next(c for c in module_registry.checks["module"] if c.id == "CKV_TF_MODULE_2")
        module_registry.checks["module"].remove(check)
```

**Review by gruebel — COMMENTED** [review]
you have to remove the added checks, otherwise they will flag resources in other tests

**Pull request merged**